keymanagementscenarios.md

```diff
@@ -0,0 +1,154 @@
+    - Signing host generates signatures with keys in configured location.
+- Signing in Air Gapped Environments with Shared Root 
+    - Publisher Admin creates new root, uploads root public key to registry, and shares root public key (similar justification to earlier scenarios).
+    - Publisher Admin creates new subordinate key in air gapped environment on a host, USB Token, Network HSM, or Cloud Key Management Service chaining to root (one time operation enables the root to be used by runtime environments outside of the air gapped environment to validate signatures generated inside the environment).
+    - Publisher Admin configures credentials for Publisher Signers in air gapped environment to use sub-ordinate key for delegate keys instead of root key (users in air gapped environment do not need to get keys from outside of the environment). 
+
+Deployment Use Cases:
+
+Provisioning Truststore for Deployment
+
+![Notaryv2 Deployment Provisioning](./media/nv2-keymgmt-architecture-provisiontruststore-v1.png)
+
+Deployment Workflow
+
+![Notaryv2 Deployment Workflow](./media/nv2-keymgmt-architecture-pull-v1.png)
+
+Detailed Usecases:
+- Specify trusted public root keys
+    - Deployer Admin gets root public key from publisher.
+    - Deployer Admin adds root public key to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Specify location of trusted public root keys
+    - Deployer Admin gets location of key repository from publisher.
+    - Deployer Admin adds location of key repository to runtime configuration.
+    - Container pulled from any registry is be validated with listed root public keys before execution.
+- Enterprises can restrict users to add/remove their root public keys to the trust store used across a fleet of runtime hosts. The trust store will validate images pulled from any registry.
+- Enterprises can restrict users to add/remove root public keys for trusted third parties to their trust stores. The trust store will validate images pulled from any registry.
+- Air gapped environments can either specify the 
+
+Key Distribution Workflows:
+- Manual Configuration
+    - Publisher places their root certificate in a public location. This could be a website similar to what public CAs today provide.
+    - Developer copies certificate from public location and adds it to their trust store. Developer can configure whether they trust this certificate for all their artifacts, individual registries, individual repositories, or individual targets.
+- Automated Configuration
+    - Publisher gets intermediate certificate from a key distriution service i.e public CA or registry operator.
+    - Developer adds root certificate and location of key respoitory to their trust store. Developer can configure whether they trust certificates for all their artifacts, individual registries, individual repositories, or individual targets.
```

Comment:
Can the public CA / registry operator restrict the scope of the intermediate certificate (to a single host name or a namespace on the operator’s registry)? If not, a single registry customer could sign images for any other customer of that registry.

(And if the scope is restricted, we need to consider how that interacts with moved images in disconnected environments.)

Reply:
The scope for signing is not restricted. This is addressed on signature validation. The deployer can specify which intermediates they trust for which registry/repository/target.

Reply:
I can’t see how that works. If I trust the Docker CA for docker.io, that doesn’t mean I trust _every Docker customer_ to sign docker.io/Contoso, I trust only the Contoso Docker customer to sign docker.io/Contoso.

If I have to specify the Contoso public key for docker.io/Contoso to have usable security, and figure out which is the right public key without relying on the Docker CA (OTOH the “trust store certificate” does mention “Identity”, so that might be possible), the Docker CA serves no purpose.